ocp_resources/plan.py

```diff
@@ -26,38 +26,24 @@ class Plan(NamespacedResource, MTV):
 
     def __init__(
         self,
-        name=None,
-        namespace=None,
-        source_provider_name=None,
-        source_provider_namespace=None,
-        destination_provider_name=None,
-        destination_provider_namespace=None,
-        storage_map_name=None,
-        storage_map_namespace=None,
-        network_map_name=None,
-        network_map_namespace=None,
-        virtual_machines_list=None,
-        target_namespace=None,
-        warm_migration=False,
-        pre_hook_name=None,
-        pre_hook_namespace=None,
-        after_hook_name=None,
-        after_hook_namespace=None,
-        client=None,
-        teardown=True,
-        yaml_file=None,
-        delete_timeout=TIMEOUT_4MINUTES,
+        source_provider_name: str | None = None,
```

Comment:
why not use the same format we use for class generator?

Reply:
what do you mean?

Reply:
Using Optional and for strings would be
`source_provider_name: Optional[str] = ""`

Reply:
`|` is the recommended way.
we should also change class-generator to use this.

Reply:
Need to add `from __future__ import annotations`